### PR TITLE
Add test to verify canonical IDs are stable

### PR DESCRIPTION
**Pull request by schmidt-sebastian**
There is a risk in the Protobuf FieldValue rewrite, so let's add a test to make sure we don't break the IDs.

### CODE REVIEW

File: firebase-firestore/src/test/java/com/google/firebase/firestore/core/QueryTest.java [hunk 1]
```diff
@@ -553,4 +559,62 @@ public void testMatchesAllDocuments() {
...
+
+  @Test
+  public void testCanonicalIdsAreStable() {
+    // This test aims to ensure that we do not break canonical IDs, as they are used as keys in
+    // the TargetCache.
+
+    Query baseQuery = Query.atPath(ResourcePath.fromString("collection"));
+
+    Map<Query, String> queryToCanoncialIdMap = new LinkedHashMap<>();
+    queryToCanoncialIdMap.put(baseQuery, "collection|f:|ob:__name__asc|lt:LIMIT_TO_FIRST");
```
Comment (wilhuff) [inline firebase-firestore/src/test/java/com/google/firebase/firestore/core/QueryTest.java]: Consider adding an assertion helper instead. Something like

```java
private void assertCanonicalId(String canonicalId, Query query) {
  assertEquals(canonicalId, query.getCanonicalId());
}
```
These end up being about the same size at the call site but are easier to step through/debug/see exactly what failed just from exception line numbers.

Comment (schmidt-sebastian) [reply firebase-firestore/src/test/java/com/google/firebase/firestore/core/QueryTest.java]: Done

File: firebase-firestore/src/test/java/com/google/firebase/firestore/core/QueryTest.java [hunk 2]
```diff
@@ -553,4 +558,67 @@ public void testMatchesAllDocuments() {
     query = baseQuery.endAt(new Bound(Collections.emptyList(), true));
     assertFalse(query.matchesAllDocuments());
   }
+
+  @Test
+  public void testCanonicalIdsAreStable() {
+    // This test aims to ensure that we do not break canonical IDs, as they are used as keys in
```
Comment (wu-hui) [inline firebase-firestore/src/test/java/com/google/firebase/firestore/core/QueryTest.java]: Sorry
It's the `Target`'s canonical ID that is used in Target cache. `Query` canonical ID is only used in `SpecTestCase` and I have a TODO to remove it all together.

Comment (schmidt-sebastian) [reply firebase-firestore/src/test/java/com/google/firebase/firestore/core/QueryTest.java]: Done. It makes me wish we had a test class for Target though.

Comment (wu-hui) [reply firebase-firestore/src/test/java/com/google/firebase/firestore/core/QueryTest.java]: Yeah, we probably should..

**Pull request merged**